content/oci/oci.go

```diff
@@ -156,16 +180,35 @@ func (s *Store) Delete(ctx context.Context, target ocispec.Descriptor) error {
 			untagged = true
 		}
 	}
-	if err := s.graph.Remove(ctx, target); err != nil {
-		return err
-	}
+	danglings := s.graph.Remove(target)
 	if untagged && s.AutoSaveIndex {
 		err := s.saveIndex()
 		if err != nil {
-			return err
+			return nil, err
 		}
 	}
-	return s.storage.Delete(ctx, target)
+	if err := s.storage.Delete(ctx, target); err != nil {
+		return nil, err
+	}
+	return danglings, nil
+}
+
+func (s *Store) autoGCInDelete(ctx context.Context, danglings []ocispec.Descriptor) error {
+	// for each item in dangling, if it exists and it is dangling, remove it and
+	// add new dangling nodes into dangling
+	i := 0
+	for i < len(danglings) {
+		node := danglings[i]
```

Comment:
You are actually using `danglings` as a queue. Here's a more elegant implementation of an array-based queue in golang:
```suggestion
	for len(danglings) > 0 {
		node := danglings[0]
		danglings = danglings[1:]
```

Reply:
changed accordingly.